ui/pages/confirmations/confirmation/confirmation.js

```diff
@@ -514,7 +514,16 @@ export default function ConfirmationPage({
           </Box>
         ) : null}
         {useSnapHeader && (
-          <SnapAuthorshipHeader snapId={pendingConfirmation?.origin} />
+          <SnapAuthorshipHeader
+            snapId={pendingConfirmation?.origin}
+            onCancel={
+              templatedValues.onCancel ||
+              // /!\ Treat cancel as submit only if approval type is appropriate /!\
+              (pendingConfirmation?.type === ApprovalType.SnapDialogAlert
+                ? handleSubmit
+                : null)
+            }
```

Comment:
IMO this should reject the confirmation by using the `rejectApproval` action rather than using the templated values

Reply:
I'm not quite sure how that can be done. Do you have an example of it or maybe a solution proposal on how it can be used/integrated into the confirmation page?

Reply:
This action is accessible from redux, I think you'll just have to import and dispatch the action ?